tests/by-util/test_tr.rs

```diff
@@ -292,3 +291,129 @@ fn test_more_than_2_sets() {
+    new_ucmd!()
+        .args(&["abc[:alpha:]", "xyz"])
+        .pipe_in("abcdefghijklmnopqrstuvwxyz")
+        .succeeds()
+        .stdout_is("zzzzzzzzzzzzzzzzzzzzzzzzzz");
+}
+
+#[test]
+fn alnum_overrides_translation_to_fallback_2() {
+    // echo -n "abcdefghijklmnopqrstuvwxyz" | tr "[:alpha:]abc" "xyz"
+    new_ucmd!()
+        .args(&["[:alpha:]abc", "xyz"])
+        .pipe_in("abcdefghijklmnopqrstuvwxyz")
+        .succeeds()
+        .stdout_is("zzzzzzzzzzzzzzzzzzzzzzzzzz");
+}
+
+#[test]
+fn overrides_translation_pair_if_repeats() {
+    // echo -n 'aaa' | tr "aaa" "xyz"
+    new_ucmd!()
+        .args(&["aaa", "xyz"])
+        .pipe_in("aaa")
+        .succeeds()
+        .stdout_is("zzz");
+}
+
+#[test]
+fn uppercase_conversion_works_1() {
+    // echo -n 'abcdefghijklmnopqrstuvwxyz' | tr "abcdefghijklmnopqrstuvwxyz" "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    new_ucmd!()
+        .args(&["abcdefghijklmnopqrstuvwxyz", "ABCDEFGHIJKLMNOPQRSTUVWXYZ"])
+        .pipe_in("abcdefghijklmnopqrstuvwxyz")
+        .succeeds()
+        .stdout_is("ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+}
+
+#[test]
+fn uppercase_conversion_works_2() {
+    // echo -n 'abcdefghijklmnopqrstuvwxyz' | tr "a-z" "A-Z"
+    new_ucmd!()
+        .args(&["a-z", "A-Z"])
+        .pipe_in("abcdefghijklmnopqrstuvwxyz")
+        .succeeds()
+        .stdout_is("ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+}
+
+#[test]
+fn uppercase_conversion_works_3() {
+    // echo -n 'abcdefghijklmnopqrstuvwxyz' | tr "[:lower:]" "[:upper:]"
+    new_ucmd!()
+        .args(&["[:lower:]", "[:upper:]"])
+        .pipe_in("abcdefghijklmnopqrstuvwxyz")
+        .succeeds()
+        .stdout_is("ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+}
+
+#[test]
+fn translate_complement_set_in_order() {
+    // echo -n '01234' | tr -c '@-~' ' -^'
+    new_ucmd!()
+        .args(&["-c", "@-~", " -^"])
+        .pipe_in("01234")
+        .succeeds()
+        .stdout_is("PQRST");
+}
+
+#[test]
+fn alpha_expands_uppercase_lowercase() {
+    // echo -n "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz" | tr "[:alpha:]" " -_"
+    new_ucmd!()
+        .args(&["[:alpha:]", " -_"])
+        .pipe_in("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")
+        .succeeds()
+        .stdout_is(r##" !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRS"##);
+}
+
+#[test]
+fn alnum_expands_number_uppercase_lowercase() {
+    // echo -n "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz" | tr "[:alnum:]" " -_"
+    new_ucmd!()
+        .args(&["[:alnum:]", " -_"])
+        .pipe_in("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")
+        .succeeds()
+        .stdout_is(r##" !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]"##);
+}
+
+#[test]
+#[ignore = "not expected to fully pass -- any help appreciated!"]
+fn check_against_gnu_tr_tests() {
+    // echo -n "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz" | tr "[:alnum:]" " -_"
+    new_ucmd!()
+        .args(&["abcd", "[]*]"])
```

Comment:
So, I think the bug here is that in `Sequence::dissolve` sometimes a `std::iter::repeat` is created, which will repeat its item infinitely. When you then try to collect all items, it will fail, becuase there are literally an infinite amount of items.

Reply:
Yea....not sure how to fix it yet. Another thing that isn't handled yet is that certain sequences are not allowed in SET1 or SET2. I think I will just create a function that accepts 2 `Vec<Sequence>` and returns a translation map between the two. Then we can just put all the weird logic in there...I will try and see this weekend.

Reply:
Sounds good! That should work